From 7985f1e8fcecdf394f70147ed4c211abc84fe60c Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@freescale.com>
Date: Tue, 20 Nov 2012 15:09:26 -0600
Subject: [PATCH 107/310] ENGR00235066 - Need to add audio HAL for JB

Updated audio policy to include bluetooth and usb.
Corrected mixer paths to allow kernel to switch haedphone on/off.

Signed-off-by: Oliver Brown <oliver.brown@freescale.com>
---
 imx53_smd/audio_policy.conf | 44 +++++++++++++++++++++++++++++++++++++++-----
 imx53_smd/mixer_paths.xml   | 11 ++++-------
 imx5x/imx5x.mk              |  1 +
 3 files changed, 44 insertions(+), 12 deletions(-)

diff --git a/imx53_smd/audio_policy.conf b/imx53_smd/audio_policy.conf
index e350504..6ec67b1 100644
--- a/imx53_smd/audio_policy.conf
+++ b/imx53_smd/audio_policy.conf
@@ -5,7 +5,7 @@
 global_configuration {
   attached_output_devices AUDIO_DEVICE_OUT_EARPIECE|AUDIO_DEVICE_OUT_SPEAKER
   default_output_device AUDIO_DEVICE_OUT_SPEAKER
-  attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC
+  attached_input_devices AUDIO_DEVICE_IN_BUILTIN_MIC|AUDIO_DEVICE_IN_BACK_MIC
 }
 
 # audio hardware module section: contains descriptors for all audio hw modules present on the
@@ -18,6 +18,7 @@ global_configuration {
 # or input stream category.
 # The "channel_masks", "formats", "devices" and "flags" are specified using strings corresponding
 # to enums in audio.h and audio_policy.h. They are concatenated by use of "|" without space or "\n".
+
 audio_hw_modules {
   primary {
     outputs {
@@ -25,16 +26,49 @@ audio_hw_modules {
         sampling_rates 44100
         channel_masks AUDIO_CHANNEL_OUT_STEREO
         formats AUDIO_FORMAT_PCM_16_BIT
-        devices AUDIO_DEVICE_OUT_EARPIECE|AUDIO_DEVICE_OUT_SPEAKER|AUDIO_DEVICE_OUT_WIRED_HEADSET|AUDIO_DEVICE_OUT_WIRED_HEADPHONE|AUDIO_DEVICE_OUT_ALL_SCO|AUDIO_DEVICE_OUT_AUX_DIGITAL|AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET
+        devices AUDIO_DEVICE_OUT_EARPIECE|AUDIO_DEVICE_OUT_SPEAKER|AUDIO_DEVICE_OUT_WIRED_HEADSET|AUDIO_DEVICE_OUT_WIRED_HEADPHONE|AUDIO_DEVICE_OUT_AUX_DIGITAL
         flags AUDIO_OUTPUT_FLAG_PRIMARY
       }
+      hdmi {
+        sampling_rates dynamic
+        channel_masks dynamic
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_AUX_DIGITAL
+        flags AUDIO_OUTPUT_FLAG_DIRECT
+      }
     }
     inputs {
       primary {
-        sampling_rates 8000|16000
-        channel_masks AUDIO_CHANNEL_IN_MONO
+        sampling_rates 8000|11025|16000|22050|24000|32000|44100|48000
+        channel_masks AUDIO_CHANNEL_IN_MONO|AUDIO_CHANNEL_IN_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_BUILTIN_MIC|AUDIO_DEVICE_IN_WIRED_HEADSET|AUDIO_DEVICE_IN_BACK_MIC|AUDIO_DEVICE_IN_USB_MIC
+      }
+    }
+  }
+  a2dp {
+    outputs {
+      a2dp {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_ALL_A2DP
+      }
+    }
+  }
+  usb {
+    outputs {
+      usb_accessory {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_USB_ACCESSORY
+      }
+      usb_device {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
         formats AUDIO_FORMAT_PCM_16_BIT
-        devices AUDIO_DEVICE_IN_BUILTIN_MIC
+        devices AUDIO_DEVICE_OUT_USB_DEVICE
       }
     }
   }
diff --git a/imx53_smd/mixer_paths.xml b/imx53_smd/mixer_paths.xml
index 4a8cc57..18786b7 100644
--- a/imx53_smd/mixer_paths.xml
+++ b/imx53_smd/mixer_paths.xml
@@ -1,18 +1,15 @@
 <mixer>
   <!-- These are the initial mixer settings -->
-    <ctl name="Speaker Function" value="on" />
-    <ctl name="Jack Function" value="on" />
   <ctl name="PCM Playback Volume" value="160" />
   <path name="speaker">
-      <ctl name="Speaker Function" value="on" />
+      <!-- <ctl name="Speaker Function" value="on" /> -->
       <ctl name="PCM Playback Volume" value="160" />
   </path> 
   <path name="main-mic">
-      <ctl name="Speaker Function" value="on" />
-      <ctl name="PCM Playback Volume" value="160" />
+      <ctl name="Mic Volume" value="2" />
   </path> 
   <path name="headphone">
-      <ctl name="Speaker Function" value="on" />
-      <ctl name="PCM Playback Volume" value="160" />
+      <!-- <ctl name="Speaker Function" value="off" /> -->
+      <ctl name="Headphone Playback Volume" value="103" />
   </path> 
 </mixer>
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 5c6ba1d..fde82fb 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -58,6 +58,7 @@ PRODUCT_PACKAGES += \
 #audio related lib
 PRODUCT_PACKAGES += \
 	audio.primary.imx5x			\
+	audio.usb.default			\
 	tinyplay				\
 	tinycap					\
 	tinymix					\
-- 
1.8.0

